File: desert/oasis.lua
loop
    local yak = self:findNearestEnemy()
    if yak then
        if self:distanceTo(yak) < 10 then
            self:moveXY(self.pos.x - 10, self.pos.y)
        else
            self:moveXY(self.pos.x + 10, self.pos.y)
        end
    else
        self:moveXY(self.pos.x + 10, self.pos.y)
    end
end

File: desert/oasis.clj
(def y 30)
(while true
  (def e (.findNearestEnemy this))
  (if e
    (if (and (< (.distanceTo this e) 12) (>= (+ (.x (.pos e)) 5) (.x (.pos this))))
      (.moveXY this (- (.x (.pos e)) 5) y)
      (.moveXY this (+ (.x (.pos this)) 10) y))
    (.moveXY this (+ (.x (.pos this)) 10) y)))
